spoolsense_installer/spoolman.py

```
@@ -3,51 +3,130 @@
+
+
+def _wait_for_spoolman(spoolman_url: str) -> bool:
+    """Poll Spoolman until it responds, or a ~45s budget is exhausted.
+
+    Returns True once reachable, False if it never comes up. Handles the common
+    case where Spoolman is not fully started at the moment the installer runs.
+    """
+    delays = [1, 2, 4, 8, 15, 15]  # cumulative ~45s across 6 attempts
+    for i, delay in enumerate(delays):
+        try:
+            req = urllib.request.Request(f"{spoolman_url}/api/v1/field/spool")
```

Ruff (0.15.20)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)